Tasks/JenkinsDownloadArtifacts/jenkinsdownloadartifacts.ts

```diff
@@ -54,25 +58,76 @@ function getLastSuccessful(serverEndpointUrl: string, jobName: string, cred: Cre
     const lastSuccessfulUrl: string = `${serverEndpointUrl}/job/${jobName}/api/json?tree=lastSuccessfulBuild[id,displayname]`;
 
     getRequest(lastSuccessfulUrl, cred, strictSSL).then((result) => {
-            if (result && result['lastSuccessfulBuild']) {
-                const lastSuccessfulBuildId = result['lastSuccessfulBuild']['id'];
-                if (lastSuccessfulBuildId) {
-                    defer.resolve(lastSuccessfulBuildId);
-                } else {
-                    defer.reject(new Error(tl.loc('CouldNotGetLastSuccessfuilBuildNumber')));
-                }
+        if (result && result['lastSuccessfulBuild']) {
+            const lastSuccessfulBuildId = result['lastSuccessfulBuild']['id'];
+            if (lastSuccessfulBuildId) {
+                defer.resolve(lastSuccessfulBuildId);
             } else {
                 defer.reject(new Error(tl.loc('CouldNotGetLastSuccessfuilBuildNumber')));
             }
-        })
+        } else {
+            defer.reject(new Error(tl.loc('CouldNotGetLastSuccessfuilBuildNumber')));
+        }
+    })
         .fail((err) => { defer.reject(err); });
 
     return defer.promise;
 }
 
+async function getArtifactsFromUrl(artifactQueryUrl: string, strictSSL: boolean, localPathRoot: string, itemPattern: string, handler: handlers.BasicCredentialHandler, variables: { [key: string]: any }) {
+    console.log(tl.loc('Downloading Artifacts from ArtifactDownloadUrl', artifactQueryUrl));
+
+    var templatePath = path.join(__dirname, 'jenkins.handlebars.txt');
+    var webProvider = new providers.WebProvider(artifactQueryUrl, templatePath, variables, handler, { ignoreSslError: !strictSSL });
+    var localFileProvider = new providers.FilesystemProvider(localPathRoot);
+
+    var downloaderOptions = new engine.ArtifactEngineOptions();
+    downloaderOptions = configureDownloaderOptions(downloaderOptions);
+    var downloader = new engine.ArtifactEngine();
+    await downloader.processItems(webProvider, localFileProvider, downloaderOptions);
+}
+
+async function getArtifactsFromAzureRMStorage(localPathRoot, containerName, commonVirtualPath) {
+    console.log(tl.loc('Downloading Artifacts from Azure blob storage, Container Name: ', containerName));
+
+    var connectedService: string = tl.getInput("ConnectedServiceNameARM");
```

Comment:
Create a separate class which defines all task parameters. And then you can just pass/store that parameters object.

Reply:
Did not go with that approach as all the inputs are not required to be read always, so i thought fetching inputs when required is more viable.